### ISSUE TITLE
pg_pathman become stuck while partitioning (range) table by hours or minutes.

### ISSUE THREAD

**Issue by kiselevyav** [postgrespro/pg_pathman#115]
My env:

```
 get_pathman_lib_version 
-------------------------
 10402

PG version - 9.6.4
```

My test case:

```
CREATE TABLE journal (
    id      SERIAL,
    dt      TIMESTAMP NOT NULL,
    level   INTEGER,
    msg     TEXT);

INSERT INTO journal (dt, level, msg)
SELECT g, random() * 6, md5(g::text)
FROM generate_series('2015-01-01'::date, '2015-01-02'::date, '1 minute') as g;

SELECT create_range_partitions('journal', 'dt', '2015-01-01'::date, '1 hour'::interval); // stuck at this line
```

Is it a bug or pg_pathman doesn't support partition by something less than a day?

**Comment by funbringer** [comment]
Hi @kiselevyav 

Thanks for the feedback!

> Is it a bug or pg_pathman doesn't support partition by something less than a day?

It's not _exactly_ a bug, and there's no such limitation. It's the type of the value (`'2015-01-01'::date`) that you use as 3rd argument of `create_range_partitions()`.

We use the following piece of code in order to calculate the amount of partitions:

```plpgsql
WHILE cur_value <= max_value
		LOOP
			cur_value := cur_value + p_interval;
			p_count := p_count + 1;
		END LOOP;
```

Here's the problem: `cur_value` is a `date`. Although `cur_value + p_interval` is a `timestamp`, it will be cast to `date`, and your hours/minutes/seconds will be lost, thus causing an endless loop. You should pass `'2015-01-01'::timestamp`, not `'2015-01-01'::date`.

Unfortunately, our public API is not perfect, and it's hard to improve this particular piece of code. We have to use the polymorphic `anyelement` type in order to accept literally any types, which means that we can't cast the passed value to a column's type and store the result.

Perhaps we can do something about this issue, but at the moment there's not much we can do, sorry.

**Comment by funbringer** [comment]
UPDATE: At very least we can detect endless loops, we'll add a fix for that in 1.5.

**Comment by kiselevyav** [comment]
Thanks for your assistance! Everything works excellent with ::interval data type for start_value
